contracts/ethereum_contracts.go

```diff
@@ -23,6 +24,39 @@ type EthereumFluxAggregator struct {
 	address        *common.Address
 }
 
+// FilterRoundSubmissions filters rounds submissions, if there isn't enogh submissions found returns and error
+func (f *EthereumFluxAggregator) FilterRoundSubmissions(ctx context.Context, submissions int, submissionVal *big.Int, roundID int) (int64, error) {
```

Comment:
Logic is too specific within a contract function, it should be a wrapper client. Test logic should handle whether there's enough submissions etc.